text/trace/0174-http-semantic-conventions.md

```diff
@@ -0,0 +1,166 @@
+and thus to ship and use stable instrumentation.
+
+> NOTE. This OTEP captures a scope for changes should be done to existing
+experimental semantic conventions for HTTP, but does not propose solutions.
+
+## Roadmap for v1.0
+
+1. This OTEP, consisting of scenarios/open questions and a proposed roadmap, is
+   approved and merged.
+2. [Stability guarantees](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#not-defined-semantic-conventions-stability)
+   for semantic conventions are approved and merged. This is not strictly related
+   to semantic conventions for HTTP but is a prerequisite for stabilizing any
+   semantic conventions.
+3. Separate PRs addressing the scenarios and open questions listed in this
+   document are approved and merged.
+4. Proposed specification changes are verified by prototypes for the scenarios
+   and examples below.
+5. The [specification for HTTP semantic conventions for tracing](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md)
+   are updated according to this OTEP and are declared
+   [stable](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#stable).
+
+The steps in the roadmap don't necessarily need to happen in the given order,
+some steps can be worked on in parallel.
+
+## Scope for v1.0: scenarios and open questions
+
+> NOTE. The scope defined here is subject for discussions and can be adjusted
+  until this OTEP is merged.
+
+Scenarios and open questions mentioned below must be addressed via separate PRs.
+
+### Required attribute sets
+
+> At least one of the following sets of attributes is required:
+>
+> * `http.url`
+> * `http.scheme`, `http.host`, `http.target`
+> * `http.scheme`, [`net.peer.name`](span-general.md), [`net.peer.port`](span-general.md), `http.target`
+> * `http.scheme`, [`net.peer.ip`](span-general.md), [`net.peer.port`](span-general.md), `http.target`
+
+As a result, users that write queries against raw data or Zipkin/Jaeger don't
+have consistent story across instrumentations and languages. e.g. they'd need to
+write queries like
+`select * where (getPath(http.url) == "/a/b" || getPath(http.target) == "/a/b")`
+
+### Retries and redirects
+
+Each try/redirect/hedging request must have unique context to be traceable and
+to unambiguously ask for support from downstream service, which implies span
+per call.
+
+Redirects: users may need observability into what server hop had an error/took
+too long. E.g., was 500/timeout from the final destination or a proxy?
+
+### Context propagation needs explanation
+
+* Reusing instances of client HTTP requests between tries (it’s likely, so clean
+  up context before making a call).
+
+### Security concerns
```

Comment:
We should explicitly call out attributes here that have potential to contain sensitive information (PII, UGC, heaven forbid auth values).

In the past, we've had issues we services:
- Leaking PIU as part of the URI (either as part of the path or the query string)
- Exposing the IP address of the client
- Including usernames as part of attributes 

With attributes that have potential to expose sensitive information, we can address in future conventions with a broader scope on how it should be handled.

Reply:
Yes, I agree that as a first step we can extend attribute [data model](https://github.com/open-telemetry/opentelemetry-specification/blob/main/semantic_conventions/trace/http.yaml) to add `potentially_sensitive=true/false` property (or similar). Going forward, we might want to come up with an approach on how to allow end users to sanitize/remove that data. Added this to the OTEP. @cobly please feel free to link this OTEP from the issue/PR addressing this for v1.

Reply:
We are currently going through an internally draft at the moment to bounce ideas of each other, however, I'll raise the OTEP by the end of next week at the latest.

Reply:
^ https://github.com/open-telemetry/oteps/pull/187